compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -941,7 +941,7 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
     val Apply(qual, args) = tree
 
     def notAnExtractor(tree: Tree) =
-      if (tree.tpe.isErroneous) tree
+      if (!tree.tpe.isError && tree.tpe.isErroneous) tree
```

Comment:
Why the special case for `isError` here ?

Reply:
`isError = true` implies an error on the outer-most tree, thus `NotAnExtractor` error should be preferred.

Reply:
> thus NotAnExtractor error should be preferred

Did you mean "should _not_ be preferred" ? You're not using the error from NotAnExtractor when isError is true.

Reply:
The check `!tree.tpe.isError && tree.tpe.isErroneous ` ensures that `isError = true` will go to the 2nd branch, which is a `NotAnExtractor` error.

Reply:
Ah right, got confused, but can you add a comment that gives an example where NotAnExtractor should be preferred to get a good error message ?